examples/repo_example/succinct_hash_bin_delegations.py

```diff
@@ -0,0 +1,190 @@
+bin delegation, but the delegating metadata is smaller, resulting in fewer bytes
+to transfer and parse.
+
+See 'basic_repo.py' for a more comprehensive TUF metadata API example.
+
+For a comprehensive explanation of succinct hash bin delegation and the
+difference between succinct and standard hash bin delegation read:
+https://github.com/theupdateframework/taps/blob/master/tap15.md
+
+NOTE: Metadata files will be written to a 'tmp*'-directory in CWD.
+"""
+import math
+import os
+import tempfile
+from datetime import datetime, timedelta
+from pathlib import Path
+from typing import Dict, Tuple
+
+from securesystemslib.keys import generate_ed25519_key
+from securesystemslib.signer import SSlibSigner
+
+from tuf.api.metadata import (
+    Delegations,
+    Key,
+    Metadata,
+    SuccinctRoles,
+    TargetFile,
+    Targets,
+)
+from tuf.api.serialization.json import JSONSerializer
+
+# Succinct hash bin delegation
+# ============================
+# Succinct hash bin delegation aims to distribute a large number of target files
+# over multiple delegated targets metadata roles (bins). The consequence is
+# smaller metadata files and thus a lower network overhead for repository-client
+# communication.
+#
+# The assignment of target files to a target's metadata is done automatically,
+# based on the byte digest of the target file name.
+#
+# The number of bins, name prefix for all bins and key threshold are all
+# attributes that need to be configured.
+
+# Number of bins, bit length and bin number computation
+# -----------------------------------------------------
+# Determining the correct number of bins is dependent on the expected number of target files in a
+# repository. For the purpose of this example we choose:
+NUMBER_OF_BINS = 32
+#
+# The number of bins will determine the number of bits in a target path considered in assigning the target to a bin.
+# for each target path. It can be calculated directly from NUMBER_OF_BINS:
+BIT_LENGTH = int(math.log2(NUMBER_OF_BINS))
+
+# Delegated role (bin) name format
+# --------------------------------
+# Each bin has a name in the format of f"{NAME_PREFIX}-{bin_number}".
+#
+# Name prefix is the common prefix of all delegated target roles (bins).
+# For our example it will be:
+NAME_PREFIX = "delegated_bin"
+#
+# The suffix "bin_number" is a zero-padded hexadecimal number of that
+# particular bin.
+
+# Keys and threshold
+# ------------------
+# Succinct hash bin delegation uses the same key(s) to sign all bins. This is
+# acceptable because the primary concern of this type of delegation is to reduce
+# network overhead. For the purpose of this example only one key is required.
+THRESHOLD = 1
+
+
+def create_key() -> Tuple[Key, SSlibSigner]:
+    """Generates a new Key and Signer."""
+    sslib_key = generate_ed25519_key()
+    return Key.from_securesystemslib_key(sslib_key), SSlibSigner(sslib_key)
+
+
+key, signer = create_key()
+
+# Top level targets instance with succinct hash bin delegation
```

Comment:
Btw. the delegator does not have to be the top-level targets role. I think we can just omit that piece of information.

Reply:
If you don't mind, I'll also try to shorten the comments in this and the next section a bit.

Reply:
Addressed in 50960c40761a0624f7f2c828135cd50d9c875ee6